lesson2/README.md

```diff
@@ -1,14 +1,85 @@
+
+Activities contain one or many fragments inside of them.  It handles the passing of information between fragments.  Most simple apps (like the ones you'll be making in the beginning of this class) will only need one activity.
+
+#### [Fragments](https://developer.android.com/guide/components/fragments.html) (ex. MainActivityFragment.java)
+Fragments are basically what we described as "screens" in our Snapchat example. Each fragment does a separate task and shows different components to the user. If you were making Snapchat, you might have a fragment for each swipeable screen. Fragments are useful for:
+1. Modularity - dividing complex activity code across fragments for better organization and maintenance.
+2. Reusability - placing behavior or UI parts into fragments that can be shared across multiple activities.
+3. Adaptability - representing sections of a UI as different fragments and utilizing different layouts depending on screen orientation and size.
+
+### 2. [Layout XML](https://developer.android.com/guide/topics/ui/declaring-layout.html) files (ex. fragment_main.xml)
+XML files are basically the CSS/HTML of the Android world. They define where each component goes in your activities and fragments. Each activity and fragment will have its own XML file that it can communicate with.
+
+### 3. Configuration Files
+#### [Manifest](https://developer.android.com/guide/topics/manifest/manifest-intro.html) (AndroidManifest.xml)
+The Manifest file defines settings about your app and what permissions it needs.  Whenever you download an app from the Play Store, it will show you what permissions the app needs (like internet, data storage, GPS location) so that users can be knowledgable about what their apps are using. These permissions are taken from the app's manifest file.
 
+#### [Gradle file](https://developer.android.com/studio/build/index.html) (build.grade (Module: app))
+Gradle is the thing that builds your app, like a compiler. Each time you launch your app to your phone to test it, gradle will run and compile your code.
+
+#### strings.xml
+Android coding conventions dictate that every user-facing string should be stored in this file instead of hardcoded.  This is so every string in your app can be easily translated in one file.
+
+## Android Studio Overview
+Android Studio is an IDE, not a text editor. This means that Android Studio is very smart and you can use it to navigate and write code quickly.  Here's an overview of some of our favorite features:
+
+- **Auto-saving:** No need to ever press the save button, AS saves as soon as you type.
+- **Auto-compiling**: AS does some compiling in real-time, which means you'll know immediately if you've misspelled a variable or didn't pass the right parameters to a function.
+- **Smart Autocomplete:** AS will only make autocomplete suggestions that are legal and make sense in the context of the code.
```

Comment:
That AS suggests things related to what you type. On a second thought, it is similar to autocomplete. So I think you can ignore my previous comment